packages/back-end/src/controllers/datasources.ts

```diff
@@ -199,15 +199,20 @@ export async function deleteDataSource(
   req: AuthRequest<null, { id: string }>,
   res: Response
 ) {
-  req.checkPermissions("createDatasources");
-
   const { org } = getOrgFromReq(req);
   const { id } = req.params;
 
   const datasource = await getDataSourceById(id, org.id);
   if (!datasource) {
     throw new Error("Cannot find datasource");
   }
+  if (datasource?.projects?.length) {
+    for (const project of datasource.projects) {
+      req.checkPermissions("createDatasources", project);
```

Comment:
Would it be feasible to overload `checkPermissions` to accept a list of projects and do the iteration 'behind the scenes'?

Reply:
Sure, we can do that. Added...